### PR TITLE
Fix renderers tests for Windows

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Fixes the `test_rendering_includes` test in the `integration.renderers.test_pydsl` tests
The problem was the line endings being created in the new file on Windows. The `textwrap.dedent` function doesn't account for Windows Style line endings.
Also fixed the problem where the output file wasn't being deleted if the test failed. The `finally` block wasn't being evaluated for some reason....

Tested on LInux

### Tests written?
Yes

### CODE REVIEW

File: tests/integration/renderers/test_pydsl.py [hunk 1]
```diff
@@ -23,26 +23,40 @@ def test_rendering_includes(self):
         inability to load custom modules inside the pydsl renderers. This
         is a FIXME.
         '''
-        try:
```
Comment (cachedout) [inline tests/integration/renderers/test_pydsl.py]: Why have the try/finally blocks been removed?

Comment (twangboy) [inline tests/integration/renderers/test_pydsl.py]: It was there to delete the output file... but if the test failed, the finally block wasn't being evaluated for some reason. I don't think it was really there to catch an error...

File: tests/integration/renderers/test_pydsl.py [hunk 2]
```diff
@@ -23,26 +23,40 @@ def test_rendering_includes(self):
...
+                       'Y2 extended \r\n' \
+                       'Y3 \r\n' \
+                       'hello red 1 \r\n' \
+                       'hello green 2 \r\n' \
+                       'hello blue 3 \r\n'
+
+        with salt.utils.fopen('/tmp/output', 'r') as f:
+            ret = f.read()
+
+        os.remove('/tmp/output')
```
Comment (lorengordon) [inline tests/integration/renderers/test_pydsl.py]: Does this path even evaluate to anything valid on Windows? My simplistic test gets me "No such file or directory".

Comment (twangboy) [inline tests/integration/renderers/test_pydsl.py]: The path is created by the test suite and some files are placed in that directory.

File: tests/integration/renderers/test_pydsl.py (hunk 1 above)
Comment (twangboy) [inline tests/integration/renderers/test_pydsl.py]: When the test was failing in Windows, the file `output` wasn't being removed

Comment (twangboy) [inline tests/integration/renderers/test_pydsl.py]: @cachedout Should I add back the try/finally block?

Comment (twangboy) [inline tests/integration/renderers/test_pydsl.py]: @cachedout ^^^

**Comment by cachedout** [comment]
@twangboy Please rebase this and I'll test it. Thanks!

**Comment by twangboy** [comment]
@cachedout Rebased

**Pull request merged**